### PR TITLE
Fixed lifetime stats for wakapi users + fix #82

### LINKED ISSUE #82: Persist colors upon restarts and refreshes

### PR DESCRIPTION
**Pull request by LorenzoLeonardini**
Wakapi's API is not entirely compatible with WakaTime. In particular, the `all_time_since_today` endpoint did not return the project name, which the `LifetimeStats` class required.

Therefore, I added a "fallback" project name, so that Wakatime users keep getting it from the backend, and Wakapi users can still use this feature.

--------------

I also kind of fixed issue #82. Colors are not persisted upon restart, that fix requires changes in the CommonUtils project and it didn't feel like a good idea*. I tried using the project name hash as an index for the color, but since the space is so small there were a lot of collisions and it didn't look great. I solved it by having a final `MaterialColors` instance, instead of shuffling it every bind.

*CommonUtils' `shuffleArray` uses `ThreadLocalRandom`, which is not seedable. To have seedable randomness I would have had to change quite a few things.

### CODE REVIEW

**Review by devgianlu — APPROVED** [review]

**Comment by devgianlu** [comment]
I could probably solve the unseeded shuffle issue with a little change, do you think that would allow a better experience (aka colors propagated to the entire app)? Or is the small amount of colors the problem?

**Comment by LorenzoLeonardini** [comment]
Honestly, I think the real problem was that when you refreshed you could see the color change with no apparent reason.
The current fix solves that. I don't think it's really necessary to persist the colors even when restarting the app, simply because you don't have a memory of the old colors (at least I don't).

The small amount of colors is only relevant to using a hash as a color index. This leads to the possibility of collisions and more than one project can have the same color. Shuffling is a good way to have different colors for each project with the smallest amount of repetitions.

Basically:
- seeding is necessary to persist upon restart. The problem with seeding is that it needs changes to CommonUtils and I don't feel like it's necessary to go there
- hashing is necessary to persist even when the number of projects changes and their order with that. The problem with hashing is that the small subset of colors and the default hashing algorithm lead to collisions.

**Comment by devgianlu** [comment]
Sounds good, could you make the same change for [LineChartViewHolder](https://github.com/devgianlu/Timeless/blob/e9a9152f2161a2fe70df066453e749e3fb8c647c/app/src/main/java/com/gianlu/timeless/listing/LineChartViewHolder.java)?

**Comment by LorenzoLeonardini** [comment]
Done, also had to change the fix because turns out it wasn't actually working (don't know what happened)

**Review by devgianlu — APPROVED** [review]
